plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/EncryptedAzureBlobStoreRepositoryIntegTests.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.repositories.azure;
+
+import org.elasticsearch.common.blobstore.BlobMetaData;
+import org.elasticsearch.common.settings.MockSecureSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.LicenseService;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
+import org.elasticsearch.repositories.encrypted.DecryptionPacketsInputStream;
+import org.elasticsearch.repositories.encrypted.EncryptedRepository;
+import org.elasticsearch.repositories.encrypted.EncryptedRepositoryPlugin;
+import org.elasticsearch.repositories.encrypted.LocalStateEncryptedRepositoryPlugin;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.BeforeClass;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class EncryptedAzureBlobStoreRepositoryIntegTests extends AzureBlobStoreRepositoryTests {
+
+    private static List<String> repositoryNames;
+
+    @BeforeClass
+    private static void preGenerateRepositoryNames() {
+        List<String> names = new ArrayList<>();
+        for (int i = 0; i < 32; i++) {
+            names.add("test-repo-" + i);
+        }
+        repositoryNames = Collections.synchronizedList(names);
```

Comment:
Doe this list have to be `synchronized`? No right?

Reply:
It does, because the test methods in the class can run in parallel and there should not be two tests that use the same repository.

Reply:
Nah I don't think we can have two test methods from the same class run concurrently in the same JVM. How would that work with the single static reference to the internal node/cluster and its cleanup after every test?

Reply:
Makes sense, I don't know what made me believe otherwise.